demo/api/schema.gql

```diff
@@ -26,6 +26,8 @@ type CurrentUser {
   impersonated: Boolean
   authenticatedUser: User
   permissionsForScope(scope: JSONObject!): [String!]!
+  allowedContentScopes: [JSONObject!]!
+  allowedContentScopesWithLabels: [JSONObject!]!
```

Comment:
Why do we need to fields for this?

Reply:
ContentScopeProviderLibrary could also support the type with label in defaultValue allowing you to use `<ContentScopeProviderLibrary<ContentScope> values={user.allowedContentScopesWithLabels} defaultValue={user.allowedContentScopesWithLabels[0]}>`

Two opinions regarding api design:
- two fields for the same `allowedContentScopes` + `allowedContentScopesWithLabels` is not ideal
- JSONObject is not ideal, especially now with the addition of the label

Reply:
> Having to set the Content-Scope Type/ObjectType for the User-Permissions would add quite some complexity and this is out of scope for this PR.

agreed. Nevertheless we can discuss future plans in this PR. Regarding typing there are 2 options:

- specific type per use case (one for pageTree, one for redirects, one for dam etc): this is not suitable for UserPermissions, that's why you currently use JSON
- a general type like `Record<string, { value: string, label: string}>`: imho this is suitable for UserPermissions (for scopes with labels)

----
> Furthermore, I had to adapt to the required types for ContentScopeProviderLibrary which kind of mixes the labels into the ContentScope-Objects.

this still could be done`<ContentScopeProviderLibrary<ContentScope> values={user.allowedContentScopesWithLabels} defaultValue={user.allowedContentScopesWithLabels[0]}>`